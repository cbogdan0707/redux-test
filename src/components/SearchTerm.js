import React from "react";
import { connect } from "react-redux";

import { onInputChange } from "../actions";

export const SearchTerm = ({
  onInputChange,
  productsByCategory,
  allProducts,
  inputValue
}) => {
  if (productsByCategory) {
    return (
      <input
        type="text"
        value={inputValue || ""}
        onChange={e => onInputChange(e, productsByCategory)}
      />
    );
  }
  return (
    <input
      type="text"
      value={inputValue || ""}
      onChange={e => onInputChange(e, allProducts)}
    />
  );
};

const mapStateToProps = state => {
  return {
    productsByCategory: state.productsByCategory.filteredProducts,
    allProducts: state.products,
    inputValue: state.productsByCategory.inputValue
  };
};

export default connect(
  mapStateToProps,
  { onInputChange }
)(SearchTerm);
