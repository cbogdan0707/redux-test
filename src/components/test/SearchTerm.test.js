import React from "react";
import renderer from "react-test-renderer";

import { SearchTerm } from "../SearchTerm";

describe("Products snapshot test", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SearchTerm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
