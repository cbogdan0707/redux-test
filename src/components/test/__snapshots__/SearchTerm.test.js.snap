// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products snapshot test renders correctly 1`] = `
<input
  onChange={[Function]}
  type="text"
  value=""
/>
`;
